crates/eips/src/eip4844/sidecar.rs

```diff
@@ -228,12 +228,21 @@ where
 
 /// An error that can occur when validating a [BlobTransactionSidecar::validate].
 #[derive(Debug)]
-#[cfg(feature = "kzg")]
+#[cfg(any(feature = "c-kzg", feature = "kzg-rs"))]
 pub enum BlobTransactionValidationError {
     /// Proof validation failed.
     InvalidProof,
-    /// An error returned by [`c_kzg`].
-    KZGError(c_kzg::Error),
+    
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "c-kzg")] {
+            /// An error returned by [`c_kzg`].
+            KZGError(c_kzg::Error),
+        } else if #[cfg(feature = "kzg-rs")] {
+            /// An error returned by [`kzg-rs`].
+            KZGError(kzg_rs::KzgError),
+        }
```

Comment:
these type defs should then be isolated in a separate module and conditionally exported, so that we don't have to do this multiple times